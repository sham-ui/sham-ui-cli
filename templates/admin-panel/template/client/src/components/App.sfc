<template>
    {% import WithoutFlushLoader from 'sham-ui-without-flush-loader' %}
    {% import Layout from './layout/main.sfc' %}
    {% import PageLoadingIndicator from './PageLoadingIndicator.sht' %}

    {% if appState.tokenLoaded && appState.routerResolved %}
        <Layout/>
    {% else %}
        <WithoutFlushLoader loadingComponent=\{{PageLoadingIndicator}}/>
    {% endif %}
</template>

<script>
    import { mergeWithTemplate } from 'sham-ui-macro/babel.macro';
    import { useStorage } from '../storages/app';

    @mergeWithTemplate
    class App extends Template {

    }

    export default useStorage( 'appState' )( App );
</script>