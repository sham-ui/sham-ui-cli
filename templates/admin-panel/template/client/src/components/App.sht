{% import WithoutFlushLoader from 'sham-ui-without-flush-loader' %}
{% import Layout from './layout/main.sfc' %}
{% import PageLoadingIndicator from './PageLoadingIndicator.sht' %}

{% if tokenLoaded && routerResolved %}
    <Layout/>
{% else %}
    <WithoutFlushLoader loadingComponent=\{{PageLoadingIndicator}}/>
{% endif %}