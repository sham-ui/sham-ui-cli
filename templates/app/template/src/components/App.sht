{% import Header from './Header.sht' %}
{% import Footer from './Footer.sht' %}

<Header/>
<main>
    <pre>
        {{ description }}
    </pre>
</main>
<Footer/>
