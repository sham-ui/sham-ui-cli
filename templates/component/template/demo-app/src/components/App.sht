{% import {{ classifiedName }} from '../../../src/{{name}}' %}

<main>
    <pre>
        {{ description }}
    </pre>
    <{{ classifiedName }}/>
</main>
