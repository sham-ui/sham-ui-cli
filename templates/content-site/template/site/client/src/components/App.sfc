<template>
    {% import WithoutFlushLoader from 'sham-ui-without-flush-loader' %}
    {% import Layout from './layout/main.sfc' %}
    {% import PageLoadingIndicator from './PageLoadingIndicator.sht' %}

    {% if appState.routerResolved %}
        <Layout/>
    {% else %}
        <WithoutFlushLoader loadingComponent=\{{PageLoadingIndicator}}/>
    {% endif %}
</template>

<script>
    import { useStorage } from '../storages/app';

    export default Component( Template, useStorage( $.appState ) );
</script>
