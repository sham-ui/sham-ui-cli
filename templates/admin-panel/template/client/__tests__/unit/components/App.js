import App from '../../../src/components/App.sfc';
import renderer from 'sham-ui-test-helpers';

it( 'renders correctly', () => {
    const meta = renderer( App, {} );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
