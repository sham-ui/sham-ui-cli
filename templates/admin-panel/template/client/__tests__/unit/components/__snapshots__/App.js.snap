// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "Options": Object {
    "appState": Object {
      "routerResolved": false,
      "tokenLoaded": false,
    },
  },
  "html": "
<!--0-->
<!--0-->
<!--0-->
",
}
`;
